### ISSUE TITLE
Unable to release index locked with read_only_allow_delete

### ISSUE THREAD

**Issue by chuanzgh** [opendistro-for-elasticsearch/security#107]
I had my `.opendistro_security` index (actually, all my indices) locked-down from writes due to triggering from `cluster.routing.allocation.disk.watermark.flood_stage` setting. In layman terms, disk space full triggering a write lock on indices.

More info on this automatic setting here: https://www.elastic.co/guide/en/elasticsearch/reference/current/disk-allocator.html
> This is a last resort to prevent nodes from running out of disk space. The index block must be released manually once there is enough disk space available to allow indexing operations to continue.

Blunder aside, I found out even the `admin` user has no rights to unlock that index due to lack of permissions.

If `securityadmin.sh` is the only one that can write to that index, there seems to be a lack of support in the tool to manually unlock that index under the automatic-disk-space-triggered-lock-down circumstance.

In dev console, under `admin` user:
`PUT /.opendistro_security/_settings
{
  "index.blocks.read_only_allow_delete": null
}`
...results in:
`{
  "error": {
    "root_cause": [
      {
        "type": "security_exception",
        "reason": "no permissions for [] and User [name=admin, roles=[admin], requestedTenant=]"
      }
    ],
    "type": "security_exception",
    "reason": "no permissions for [] and User [name=admin, roles=[admin], requestedTenant=]"
  },
  "status": 403
}`
And in Elasticsearch logs:
`[2019-07-31T13:36:51,360][WARN ][c.a.o.s.p.OpenDistroSecurityIndexAccessEvaluator] [node-id] indices:admin/settings/update for '.opendistro_security' index is not allowed for a regular user`

(Originally posted @ https://github.com/opendistro-for-elasticsearch/security-kibana-plugin/issues/69 but it really should be here)

**Comment by aetter** [comment]
Hi @chuanzgh, try adding the admin certificate to the request, as shown at the very, very bottom of this page: https://opendistro.github.io/for-elasticsearch-docs/docs/elasticsearch/snapshot-restore/#security-plugin-considerations

**Comment by chuanzgh** [comment]
Thanks @aetter, this worked:
`curl -k --cert chain.pem --key kirk.key.pem -X PUT -H "Content-Type: application/json" -d '{"index.blocks.read_only_allow_delete": null }' https://localhost:9200/.opendistro_security/_settings`

**Comment by sijoyjoseph** [comment]
@aetter 
I am still getting the error:
```{"error":{"root_cause":[{"type":"security_exception","reason":"no permissions for [] and User [name=admin, roles=[admin], requestedTenant=null]"}],"type":"security_exception","reason":"no permissions for [] and User [name=admin, roles=[admin], requestedTenant=null]"},"status":403}```

this is my curl request: 
```curl --cert admin.pem --key admin-key.pem -XPUT -H "Content-type: application/json" -d '{"index.blocks.read_only_allow_delete": null}' https://localhost:9200/.opendistro_security/_settings -k```

without specifying the username and password nothing is printed in the logs. But specifying username and password prints this: ```indices:admin/settings/update for '.opendistro_security' index is not allowed for a regular user``` in the logs.
Is there something that I am missing.

**Comment by spikewang** [comment]
> @aetter
> I am still getting the error:
> `{"error":{"root_cause":[{"type":"security_exception","reason":"no permissions for [] and User [name=admin, roles=[admin], requestedTenant=null]"}],"type":"security_exception","reason":"no permissions for [] and User [name=admin, roles=[admin], requestedTenant=null]"},"status":403}`
> 
> this is my curl request:
> `curl --cert admin.pem --key admin-key.pem -XPUT -H "Content-type: application/json" -d '{"index.blocks.read_only_allow_delete": null}' https://localhost:9200/.opendistro_security/_settings -k`
> 
> without specifying the username and password nothing is printed in the logs. But specifying username and password prints this: `indices:admin/settings/update for '.opendistro_security' index is not allowed for a regular user` in the logs.
> Is there something that I am missing.

Yes, I ran into the very same problem as well... Even I used my admin cert, then I digged a bit more. The problem is your `elasticsearch.yaml`... There is one config you have to enable:

```
-    #opendistro_security.authcz.admin_dn:
-    #  - CN=admin-prod
+    opendistro_security.authcz.admin_dn:
+      - 'CN=XXX'
```

After enabling this, restarting your `Elasticsearch-master`, the above CURL with cert should just work... Cheers!
